lib/saluki-components/src/sources/checks/builder/python/python_modules.rs

```diff
@@ -67,18 +70,33 @@ fn try_send_event(event: EventD) -> Result<(), GenericError> {
 fn get_config_key(key: String) -> String {
     match GLOBAL_CONFIGURATION.get() {
         Some(configuration) => configuration.get_typed_or_default::<String>(&key),
-
         None => "".to_string(),
     }
 }
 
 fn fetch_hostname() -> &'static str {
-    match GLOBAL_HOST.get() {
-        Some(host) => host.as_str(),
+    match GLOBAL_EXECUTION_CONTEXT.get() {
+        Some(ExecutionContext { hostname, .. }) => hostname.as_str(),
         None => "",
     }
 }
 
+fn fetch_tracemalloc_enabled() -> bool {
+    match GLOBAL_CONFIGURATION.get() {
+        Some(configuration) => configuration.get_typed_or_default::<bool>("tracemalloc_debug"),
+        None => false, // FIXME: this returns a supposed default. Can we get the default without a configuration?
+    }
```

Comment:
Can we get a default with the configuration? Yeah, just set it to whatever the default is for the setting itself. :P

What I'm more interested with, overall, is how we can get away from these global statics and their associated getters. It _feels_ like it should be possible for us to define a global static on the _Python_ side which references back to a value we own whose lifetime is tied to the source component itself... but I haven't personally investigated whether this is possible and/or doable without changing the Python API that checks depend on.

Reply:
I'll try to investigate if this is possible, when time allows!